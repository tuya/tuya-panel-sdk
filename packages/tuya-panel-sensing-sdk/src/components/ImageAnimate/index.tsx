import React, { useEffect, useState, FC } from 'react';
import { StyleSheet, View, Animated, Easing, StyleProp, ViewStyle } from 'react-native';

interface ImageAnimateProps {
  style?: StyleProp<ViewStyle>;
  source?: any;
}

const ImageAnimate: FC<ImageAnimateProps> = ({ children, style, source }) => {
  const [spinValue] = useState(new Animated.Value(0));

  const spin = () => {
    spinValue.setValue(0);
    Animated.timing(spinValue, {
      toValue: 1,
      duration: 2000,
      easing: Easing.linear,
      useNativeDriver: false,
    }).start(spin);
  };

  useEffect(() => {
    spin();
  }, []);

  const rotate = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  });

  return (
    <View style={[styles.container, style]}>
      <Animated.Image
        style={[{ transform: [{ rotate }] }, styles.icon]}
        source={source || require('../../res/reload.png')}
      />
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  icon: {
    height: 16,
    marginRight: 3,
    width: 16,
  },
});
export default ImageAnimate;
