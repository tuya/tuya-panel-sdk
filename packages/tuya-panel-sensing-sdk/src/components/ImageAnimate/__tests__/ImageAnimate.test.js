// /**
//  * @jest-environment jsdom
//  */
// import React from 'react';
// import { shallow } from 'enzyme';
// import ImageAnimate from '../index';

// global.document = document;
// global.window = window;

// describe('DpCacheText', () => {
//   it('basic demo', () => {
//     const wrapper = shallow(<ImageAnimate />);
//     expect(wrapper).toMatchSnapshot();
//   });

//   it('basic demo1', () => {
//     const wrapper = shallow(
//       <ImageAnimate
//         source="https://www.google.com.hk/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png"
//         style={{ borderColor: '#fff' }}
//       />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });
// });

import React from 'react';
import { Text } from 'react-native';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ImageAnimate from '../index';

const reload =
  'https://www.google.com.hk/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png';

configure({ adapter: new Adapter() });

describe('ImageAnimate', () => {
  let useEffect;
  let effectWrapper;
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };
  beforeEach(() => {
    useEffect = jest.spyOn(React, 'useEffect');
    mockUseEffect(); // 2 times
    mockUseEffect(); //
    effectWrapper = shallow(<ImageAnimate />);
  });

  it('default render', () => {
    const wrapper = shallow(<ImageAnimate />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render widthSource', () => {
    const wrapper = shallow(<ImageAnimate source={reload} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render widthChildren', () => {
    const wrapper = shallow(
      <ImageAnimate source={reload}>
        <Text>我是children</Text>
      </ImageAnimate>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render widthStyle', () => {
    const wrapper = shallow(
      <ImageAnimate style={{ backgroundColor: 'green' }}>
        <Text>我是children</Text>
      </ImageAnimate>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render widthAll', () => {
    const wrapper = shallow(
      <ImageAnimate source={reload} style={{ backgroundColor: 'green' }}>
        <Text>我是children</Text>
      </ImageAnimate>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render width children', () => {
    const wrapper = shallow(
      <ImageAnimate source={reload} style={{ backgroundColor: 'green' }}>
        <Text>我是children</Text>
      </ImageAnimate>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('use effect', () => {
    expect(effectWrapper).toMatchSnapshot();
  });
});
