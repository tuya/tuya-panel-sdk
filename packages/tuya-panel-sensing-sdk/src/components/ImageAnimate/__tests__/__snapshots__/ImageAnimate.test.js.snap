// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageAnimate default render 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    source={1}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
</View>
`;

exports[`ImageAnimate render width children 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <AnimatedComponent
    source="https://www.google.com.hk/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png"
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
  <Text>
    我是children
  </Text>
</View>
`;

exports[`ImageAnimate render widthAll 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <AnimatedComponent
    source="https://www.google.com.hk/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png"
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
  <Text>
    我是children
  </Text>
</View>
`;

exports[`ImageAnimate render widthChildren 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    source="https://www.google.com.hk/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png"
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
  <Text>
    我是children
  </Text>
</View>
`;

exports[`ImageAnimate render widthSource 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    source="https://www.google.com.hk/images/branding/googlelogo/2x/googlelogo_color_92x30dp.png"
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
</View>
`;

exports[`ImageAnimate render widthStyle 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <AnimatedComponent
    source={1}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
  <Text>
    我是children
  </Text>
</View>
`;

exports[`ImageAnimate use effect 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    source={1}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 16,
          "marginRight": 3,
          "width": 16,
        },
      ]
    }
  />
</View>
`;
